src/content/profile-data.js

```diff
@@ -735,34 +735,36 @@ export function getSampleIndexClosestToTime(samples: SamplesTable, time: number)
   return samples.length - 1;
 }
 
-export function getJankInstances(samples: SamplesTable, processType: string, thresholdInMs: number): TracingMarker[] {
+function addTracingMarker(tracingMarkers: TracingMarker[], start, dur, title, name, data) {
+  tracingMarkers.push({start, dur, title, name, data});
+}
+
+export function getJankInstances(samples: SamplesTable, threadName: string, thresholdInMs: number): TracingMarker[] {
   let lastResponsiveness = 0;
   let lastTimestamp = 0;
   const jankInstances = [];
   for (let i = 0; i < samples.length; i++) {
     const currentResponsiveness = samples.responsiveness[i];
     if (currentResponsiveness < lastResponsiveness) {
       if (lastResponsiveness >= thresholdInMs) {
-        jankInstances.push({
-          start: lastTimestamp - lastResponsiveness,
-          dur: lastResponsiveness,
-          title: `${lastResponsiveness.toFixed(2)}ms event processing delay on ${processType} thread`,
-          name: 'Jank',
-          data: null,
-        });
+        addTracingMarker(jankInstances,
```

Comment:
actually I was thinking of a addTracingMarker arrow utility  function that would be _inside_ `getJankInstances` and get access to all the internal variables; this way it doesn't need any parameter and this would make this code more readable.

Something like:
```javascript
export function getJankInstances(samples: SamplesTable, threadName: string, thresholdInMs: number): TracingMarker[] {
  const addTrackingMarker = () => jankInstances.push({
    start: lastTimestamp - lastResponsiveness,
    dur: lastResponsiveness,
    title: `${lastResponsiveness.toFixed(2)}ms event processing delay on ${threadName}`,
    name: 'Jank',
    data: null,
  });

  let lastResponsiveness = 0;
  let lastTimestamp = 0;
  const jankInstances = [];
  for (let i = 0; i < samples.length; i++) {
    const currentResponsiveness = samples.responsiveness[i];
    if (currentResponsiveness < lastResponsiveness) {
      if (lastResponsiveness >= thresholdInMs) {
        addTracingMarker();
      }
    }
    lastResponsiveness = currentResponsiveness;
    lastTimestamp = samples.time[i];
  }
  if (lastResponsiveness >= thresholdInMs) {
    addTracingMarker();
  }
  return jankInstances;
}
```

I really think this makes the function more readable, don't you think @gregtatum ?